thorlcr/thorutil/thmem.cpp

```diff
@@ -271,14 +271,19 @@ class CSharedSpillableRowSet : public CSpillableStreamBase, implements IInterfac
         offset_t outputOffset;
         Owned<IRowStream> spillStream;
         Linked<CSharedSpillableRowSet> owner;
+        rowidx_t toRead = 0;
+        bool eos = false;
     public:
         IMPLEMENT_IINTERFACE_USING(CSimpleInterface);
 
         CStream(CSharedSpillableRowSet &_owner) : owner(&_owner)
         {
+            // NB: CStream constructor called within rows lock and only called if not yet spilled
             pos = 0;
             outputOffset = (offset_t)-1;
-            owner->rows.registerWriteCallback(*this); // NB: CStream constructor called within rows lock
+            toRead = owner->rows.numCommitted();
+            eos = (0 == toRead);
+            owner->rows.registerWriteCallback(*this);
```

Comment:
It will still be called in the dtor.

But I think I'll change it so that if 0 rows, it doesn't even create a CStream.